packages/plugins/@nocobase/plugin-data-visualization/src/client/flow/components/AntdFilterSelector.tsx

```diff
@@ -95,6 +105,56 @@ export const AntdFilterSelector: React.FC<AntdFilterSelectorProps> = ({
     return () => dispose();
   }, [onChange]);
 
+  const viewCtx = useFlowViewContext();
+  const enhancedCtx = React.useMemo(() => {
+    if (!Array.isArray(collectionPath) || collectionPath.length < 2) return null;
+    const [dataSourceKey, collectionName] = collectionPath || [];
+    const ctx = new FlowContext();
+    ctx.addDelegate(viewCtx);
+
+    model.context.defineProperty('collection', {
+      get: () => model.context.dataSourceManager.getCollection(dataSourceKey, collectionName),
+      cache: false,
+      meta: (() => {
+        const baseFactory = createCollectionContextMeta(
+          () => model.context.dataSourceManager.getCollection(dataSourceKey, collectionName),
+          model.translate('Current collection'),
+        );
+        return async () => {
+          try {
+            const base = await baseFactory();
+            if (!base) return null;
+            const raw = typeof base.properties === 'function' ? await base.properties() : base.properties;
+            const dm = model.context.app?.dataSourceManager;
+            const ds = dm?.getDataSource(dataSourceKey);
+            const cm = ds?.collectionManager;
+            const fim = dm?.collectionFieldInterfaceManager;
+            const list = await (model as unknown as { getFilterFields: () => Promise<any[]> }).getFilterFields();
+            const allow = list?.map((f: any) => f.name) || [];
+            const out: Record<string, any> = {};
+            for (const name of allow) {
+              const metaProp = raw?.[name];
+              if (!metaProp) continue;
+              const fieldDef = (cm?.getCollectionFields(collectionName) || []).find((ff: any) => ff.name === name);
+              const iface = fieldDef?.interface ? fim?.getFieldInterface(fieldDef.interface) : undefined;
+              if (iface?.filterable?.nested && fieldDef?.target) out[name] = metaProp;
+              else {
+                const clone: any = { ...metaProp };
+                delete clone.properties;
+                out[name] = clone;
+              }
+            }
+            return { ...base, properties: out };
+          } catch {
+            return null;
+          }
+        };
+      })(),
```

Comment:
```suggestion
      meta: createCollectionContextMeta(
          () => model.context.dataSourceManager.getCollection(dataSourceKey, collectionName),
          model.translate('Current collection'),
        ),
```

如果需要过滤掉不可过筛选的字段，可以直接在 createCollectionContextMeta 方法中修改。这样所有使用 createCollectionContextMeta 方法的地方都能被优化。